### PR TITLE
feat: Descriptions add vertical layout

### LINKED ISSUE #16960: Descriptions: add vertical / horizontal layout prop
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

Screenshot 2019-06-04 at 16 48 27

### What problem does this feature solve?
It would be great to control the Descriptions layout vertical, horizontal as It was possible with antd pro v2, now Descriptions have different DOM structure and quite not possible to use vertically just from modifying css

### What does the proposed API look like?

```
<Descriptions vertical>
  ...
</Descriptions>
``` 

or

```
<Descriptions layout="vertical">
  ...
</Descriptions>
```

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / document update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

close #16960 

我想贴张图的 无奈 上传不上去

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Descriptions supports layout props.         |
| 🇨🇳 Chinese |    Descriptions支持layout属性       |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/descriptions/index.zh-CN.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ cols: 1
 | bordered | 是否展示边框 | boolean | false |
 | column | 一行的 `DescriptionItems` 数量，可以写成像素值或支持响应式的对象写法 `{ xs: 8, sm: 16, md: 24}` | number | 3 |
 | size | 设置列表的大小。可以设置为 `middle` 、`small`, 或不填（只有设置 `bordered={true}` 生效） | `default | middle | small` | false |
+| layout | 描述布局 | `horizontal | vertical` | 'horizontal' |
```
Comment (afc163) [inline components/descriptions/index.zh-CN.md]: 表格渲染有点问题

File: components/descriptions/demo/vertical.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+order: 4
+title:
+  zh-CN: 垂直
+  en-US: Vertical
+---
+
+## zh-CN
+
+简单的展示。
```
Comment (afc163) [inline components/descriptions/demo/vertical.md]: 描述修改一下。

File: components/descriptions/demo/vertical-border.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+order: 5
+title:
+  zh-CN: 垂直带边框的
+  en-US: Vertical border
+---
+
+## zh-CN
+
+带边框和背景颜色列表。
```
Comment (ztplz) [inline components/descriptions/demo/vertical-border.md]: 少个`的`

**Review by ztplz — DISMISSED** [review]

File: components/descriptions/index.zh-CN.md (hunk 1 above)
Comment (hengkx) [reply components/descriptions/index.zh-CN.md]: 表格 是加上rowSpan 充满整行吗？

File: components/descriptions/demo/vertical.md (hunk 2 above)
Comment (hengkx) [reply components/descriptions/demo/vertical.md]: 改为`垂直的描述列表。`？

File: components/descriptions/index.zh-CN.md (hunk 1 above)
Comment (afc163) [reply components/descriptions/index.zh-CN.md]: 是 API 表格渲染有问题。

File: components/descriptions/demo/vertical-border.md (hunk 3 above)
Comment (hengkx) [reply components/descriptions/demo/vertical-border.md]: 改为 `垂直带边框和背景颜色的列表。`？

**Comment by hengkx** [comment]
image
image

带边框的 好像有点问题 我在调整下

**Comment by hengkx** [comment]
image

**Review by afc163 — APPROVED** [review]

**Pull request merged**